designs/node-overlay-dra.md

```diff
@@ -0,0 +1,564 @@
+| `spec.devices[].bindingFailureConditions` | Alpha (Device Binding Conditions). Shouldn't be required even when the feature goes beta / GA since it's not relevant to instance type selection.                                        |
+| `spec.devices[].bindsToNode`              | Alpha (Device Binding Conditions). Shouldn't be required even when the feature goes beta / GA since it's not relevant to instance type selection.                                        |
+| `spec.devices[].consumesCounters`         | Alpha (Consumable Capacity)                                                                                                                                                              |
+| `spec.devices[].nodeName`                 | Alpha (Partitionable Devices). Wouldn't be supported regardless due to it being an unresolvable value.                                                                                   |
+| `spec.devices[].nodeSelector`             | Alpha (Partitionable Devices)                                                                                                                                                            |
+| `spec.devices[].taints`                   | Alpha (Device Taints)                                                                                                                                                                    |
+
+New fields would be evaluated on a case-by-case basis as they are added to the upstream schema. Additionally, new fields
+may be introduced to support functionality specific to Karpenter. An example of this can be found in the section on
+[supporting relative topology requirements](#supporting-relative-topology-requirements). A complete description of the
+proposed `ResourceSliceTemplate` schema can be found in the appendix.
+
+### Supporting Relative Topology Requirements
+
+One of the main additional capabilities that DRA enables is non-node local device consumption. Drivers can define the
+set of nodes a device can be consumed from using `spec.nodeSelector` on the `ResourceSlice`. Generally speaking, this
+topology requirement is going to be **relative** to the device's "host node." Going forward, we'll consider an example
+of a device which is only available within the same zone as the host node.
+
+When a DRA driver creates `ResourceSlices` for such a device, it already knows the zone the node has been provisioned in
+and can inject that zonal requirement into the `ResourceSlice's` selector. On the other hand, the user does not
+necessarily know the zone that Karpenter will provision the node into. They could create a `NodeOverlay` per topology
+domain, but this becomes untennable with a large number of domains. This problem statement can be generalized: how does
```

Comment:
Is there a large number of topology domains?
Do most DRA resources include topology domains?

Reply:
> Do most DRA resources include topology domains?

I think it's far too early in DRA's lifecycle to make a claim one way or another. It was one of the motivating examples for DRA, and some of the motivation was to use arbitrary topology keys (like rack identifiers).